mnist-distributed/mirrored_mnist.py

```diff
@@ -0,0 +1,87 @@
+import urllib3
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+from tensorflow import keras
+from tensorflow.keras import layers
+from mlflow import log_metric
+import gzip, pickle, os
+import numpy as np
+import tensorflow as tf
+import argparse
+
+MODEL_DIR = "/model/"
+input_shape = (28, 28, 1)
+num_classes = 10
+
+def data_loader(hyperparams):
+    f = gzip.open('/mnist/mnist.pkl.gz', 'rb')
+    dataset = pickle.load(f, encoding='bytes')
+    f.close()
+    (x_train, y_train), (x_test, y_test) = dataset
+    x_train = x_train.astype("float32") / 255
+    x_test = x_test.astype("float32") / 255
+    #Make sure images have shape (28, 28, 1)
+    x_train = np.expand_dims(x_train, -1)
+    x_test = np.expand_dims(x_test, -1)
+    y_train = keras.utils.to_categorical(y_train, num_classes)
+    y_test = keras.utils.to_categorical(y_test, num_classes)
+    
+    return (
+        tf.data.Dataset.from_tensor_slices((x_train, y_train)).batch(hyperparams['BATCH_SIZE']),
+        tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(hyperparams['BATCH_SIZE']),
+    )
+    
+def model_with_strategy(learning_rate):
+    strategy = tf.distribute.MirroredStrategy()
+    with strategy.scope():
+        model = keras.Sequential(
+                [
+                    keras.Input(shape=input_shape),
+                    layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
+                    layers.MaxPooling2D(pool_size=(2, 2)),
+                    layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
+                    layers.MaxPooling2D(pool_size=(2, 2)),
+                    layers.Flatten(),
+                    layers.Dropout(0.5),
+                    layers.Dense(num_classes, activation="softmax"),
+                ]
+            )
+        model.compile(loss="categorical_crossentropy", optimizer=tf.keras.optimizers.Adam(learning_rate), metrics=["accuracy"])
+    return model
+  
+class DistributedTrainingMnistClassification(object):
```

Comment:
Rename this to DistrubtedTraining since it is generic and users can use this for non-mnist datasets

Reply:
Done